upload/src/addons/SV/ThreadReplyBanner/XF/Entity/Thread.php

```diff
@@ -5,6 +5,29 @@
 /*
  * Extends \XF\Entity\Thread
  */
+use XF\Mvc\Entity\Structure;
+
 class Thread extends XFCP_Thread
 {
+	public function canManageThreadReplyBanner(&$error = null) {
+		return \XF::visitor()->hasNodePermission(
+			$this->node_id,
+			'sv_replybanner_manage'
+		);
+	}
+
```

Comment:
Add a `canViewBanner()` method and move the logic in https://github.com/yugensoft/XenForo2-ThreadReplyBanner/blob/a92783a5258cb4dc099f96508964c4108f21c45d/upload/src/addons/SV/ThreadReplyBanner/_output/templates/public/SV_ThreadReplyBanner.html#L1-L7 into there.